tests/utils/datatype_tests.py

```diff
@@ -104,3 +104,84 @@ def test_tile_clean(self):
         string.clean(tile2, nodeid)
 
         self.assertIsNotNone(tile2.data[nodeid])
+
+
+class ReferenceDataTypeTests(ArchesTestCase):
+    def test_validate(self):
+        reference = DataTypeFactory().get_instance("reference")
+
+        for value in [
+                "", 
+                [], 
+                [{}], # reference has no 'uri'
+                [{"uri":""}], # reference uri is empty
+            ]:
+            errors = reference.validate(value="")
+            self.assertIsNotNone(errors)
```

Comment:
So that all four tests run and tell you which ones specifically failed:
```suggestion
            with self.subTest(whatever_label_you_want_in_the_output=value):
                errors = reference.validate(value="")
                self.assertIsNotNone(errors)
```